Sources/GRPCCodeGen/CodeGenerationRequest.swift

```diff
@@ -282,4 +282,39 @@ public struct CodeGenerationRequest {
       }
     }
   }
+
+  /// Represents the name associated with a namespace, service or a method, in three different formats,
+  /// which are used in specific parts of the generated code. There must be only one ``Name`` object
+  /// for each namespace, service or method.
+  ///
+  /// The base name, the generatedUpperCase (and the generatedLowerCase) must be unique for methods
+  /// from within the same service and for services within the same namespace.
+  public struct Name: Hashable {
+    /// The base name is the name used for the namespace/service/method in the IDL file, so it should follow
+    /// the specific casing of the IDL.
+    ///
+    /// The base name is also used in the descriptors that identify a specific method or service :
+    /// `<service_namespace_baseName>.<service_baseName>.<method_baseName>`.
+    public let base: String
```

Comment:
Properties on structs should be `var` unless you have a good reason for them not to be.

Making them a `let` only makes using the type harder:

```swift
var foo = Foo(bar: "bar")
foo.bar = "baz"  // Not possible if 'bar' is a 'let'
foo = Foo(bar: "baz")  // Possible, but more annoying to do than the above, especially if there are multiple properties
```